lock.go

```diff
@@ -0,0 +1,244 @@
+package ibclient
+
+import (
+	"fmt"
+	"github.com/Sirupsen/logrus"
+	"math/rand"
+	"time"
+)
+
+const (
+	timeout     int32  = 60 // in seconds
+	freeLockVal string = "Available"
+)
+
+type Lock interface {
+	Lock() bool
+	UnLock(force bool) bool
+}
+
+type NVLocker struct {
+	name          string
+	objMgr        *ObjectManager
+	LockEA        string
+	LockTimeoutEA string
+}
+
+func (l *NVLocker) createLockRequest() *MultiRequest {
+
+	req := NewMultiRequest(
+		[]*RequestBody{
+			&RequestBody{
+				Method: "GET",
+				Object: "networkview",
+				Data: map[string]interface{}{
+					"name":         l.name,
+					"*" + l.LockEA: freeLockVal,
+				},
+				Args: map[string]string{
+					"_return_fields": "extattrs",
+				},
+				AssignState: map[string]string{
+					"NET_VIEW_REF": "_ref",
+				},
+				Discard: true,
+			},
+			&RequestBody{
+				Method: "PUT",
+				Object: "##STATE:NET_VIEW_REF:##",
+				Data: map[string]interface{}{
+					"extattrs+": map[string]interface{}{
+						l.LockEA: map[string]string{
+							"value": l.objMgr.TenantID,
+						},
+						l.LockTimeoutEA: map[string]int32{
+							"value": int32(time.Now().Unix()),
+						},
+					},
+				},
+				EnableSubstitution: true,
+				Discard:            true,
+			},
+			&RequestBody{
+				Method: "GET",
+				Object: "##STATE:NET_VIEW_REF:##",
+				Args: map[string]string{
+					"_return_fields": "extattrs",
+				},
+				AssignState: map[string]string{
+					"DOCKER-ID": "*" + l.LockEA,
+				},
+				EnableSubstitution: true,
+				Discard:            true,
+			},
+			&RequestBody{
+				Method: "STATE:DISPLAY",
+			},
+		},
+	)
+
+	return req
+}
+
+func (l *NVLocker) createUnlockRequest(force bool) *MultiRequest {
+
+	getData := map[string]interface{}{"name": l.name}
+	if !force {
+		getData["*"+l.LockEA] = l.objMgr.TenantID
```

Comment:
Don't forget about this: https://github.com/infobloxopen/infoblox-go-client/pull/21#discussion_r128240666
;-)